packages/fast-usdc/src/exos/settler.js

```diff
@@ -151,17 +151,23 @@ export const prepareSettler = (
             return;
           }
 
-          if (!addressTools.hasQueryParams(tx.receiver)) {
+          let endUserDestination;
+          try {
+            const { EUD } = decodeAddressHook(tx.receiver).query;
+            if (!EUD) {
+              console.log('no EUD parameter', tx.receiver);
+              return;
+            }
+            if (!matches(EUD, M.string())) {
+              console.log('EUD is not a string', EUD);
+              return;
+            }
+            endUserDestination = /** @type {string} */ (EUD);
+          } catch (e) {
```

Comment:
In a huddle w @dckc, we discussed moving this to a helper function. Drawback is that we might lose some diagnostic info, unless we wanted to return something like `[result, errorMsg]`

Although not pleasing to my eye, this logic seems fine to keep here.